contracts/LockedNORI.sol

```diff
@@ -40,127 +63,339 @@ contract LockedNORI is ERC777Upgradeable, ERC20PresetMinterPauserUpgradeable, ER
 
     emit TokenGrantCreated(
-      recipient,
+      params.recipient,
       amount,
-      startTime,
-      vestEndTime,
-      unlockEndTime
+      params.startTime,
+      params.vestEndTime,
+      params.unlockEndTime
     );
   }
 
-  function depositedBalanceOf(address account) public view returns (uint256) {
-    return super.balanceOf(account);
+  /**
+   * @dev grantTo: Sets up an vesting + lockup schedule for recipient.
+   *
+   * Typically this will be invoked via the `tokensReceived` callback for cases
+   * where we have the tokens in hand at the time we set up the grant.
+   *
+   * It is also callable externally to handle the case that tokens are incrementally
+   * deposited after the grant is established.
+   */
+  function grantTo(
+    uint256 amount,
+    address recipient,
+    uint256 startTime,
+    uint256 vestEndTime,
+    uint256 unlockEndTime,
+    uint256 cliff1Time,
+    uint256 cliff2Time,
+    uint256 vestCliff1Amount,
+    uint256 vestCliff2Amount,
+    uint256 unlockCliff1Amount,
+    uint256 unlockCliff2Amount
+  ) public whenNotPaused onlyRole(TOKEN_GRANTER_ROLE) {
+    bytes memory userData = abi.encode(
+        recipient,
+        startTime,
+        vestEndTime,
+        unlockEndTime,
+        cliff1Time,
+        cliff2Time,
+        vestCliff1Amount,
+        vestCliff2Amount,
+        unlockCliff1Amount,
+        unlockCliff2Amount);
+    _grantTo(
+        amount,
+        userData
+    );
   }
 
-  function totalGrantAmount(address account) public view returns (uint256) {
+  /**
+   * @dev revokeUnvestedTokens: Truncates a vesting grant.
+   *
+   * Transfers any unvested tokens in *from*'s grant to *to*
+   * and reduces the total grant size.
+   *
+   * No change is made to balances that have vested but not claimed
+   * whether locked or not.
+   */
+  function revokeUnvestedTokens(
+    uint256 atTime,
+    address from,
+    address to
+  ) external onlyRole(TOKEN_GRANTER_ROLE) whenNotPaused {
+    TokenGrant storage grant = grants[from];
+    require(grant.exists, "No grant exist");
+    uint256 vestedBalance = _vestedBalanceOf(atTime, from);
+    if (vestedBalance < grant.grantAmount) {
+      uint256 quantityRevoked = grant.grantAmount - vestedBalance;
+      grant.grantAmount = vestedBalance;
+      emit UnvestedTokensRevoked(atTime, from, quantityRevoked);
+      operatorSend(
+        from,
+        to,
+        quantityRevoked,
+        "",
+        ""
+      );
+    }
+  }
+
+  /**
+   * @dev vestedBalanceOf: Vested balance less any claimed amount
+   */
+  function quantityRevokedFrom(address account) public view returns (uint256) {
     TokenGrant storage grant = grants[account];
-    return grant.grantAmount;
+    return grant.originalAmount - grant.grantAmount;
   }
 
+  /**
+   * @dev vestedBalanceOf: Vested balance less any claimed amount
+   */
   function vestedBalanceOf(address account) public view returns (uint256) {
+    return _vestedBalanceOf(block.timestamp, account);
```

Comment:
we're pretty inconsistent with this pattern of using an internal `_functionName`)

Reply:
I just updated `unlockedTokenBalance` to follow this pattern.  Is it worth doing it throughout?

Reply:
Applied the pattern to `revokeUnvestedTokens` as well.

Reply:
I don't have a strong opinion